### PR TITLE
JW Player RTD Module : populate content url, title and description

### PR DESCRIPTION
**Pull request by karimMourra**
## Type of change

- [x] Feature

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?

## Description of change
Includes content url, title and description when enriching the `ortb.site.content`.

## Other information
Docs PR: https://github.com/prebid/prebid.github.io/pull/5190

### CODE REVIEW

File: modules/jwplayerRtdProvider.js [hunk 1]
```diff
@@ -217,18 +217,40 @@ function loadVatForPendingRequest(playerDivId, mediaID, callback) {
...
+  if (!sources || !sources.length) {
+    return;
+  }
+
+  const validSource = sources.find(source => !!source.file);
+  if (!validSource) {
+    return;
+  }
+
+  return validSource.file;
```
Comment (dgirardi) [inline modules/jwplayerRtdProvider.js:251]:
```suggestion
return playlistItem.sources?.find?.(source => source.file)?.file;
```

File: modules/jwplayerRtdProvider.js [hunk 2]
```diff
@@ -217,18 +217,40 @@ function loadVatForPendingRequest(playerDivId, mediaID, callback) {
...
 
-  if (!segments) {
+  if (!item) {
     return null;
   }
 
+  let mediaUrl = item.file;
+  if (!mediaUrl) {
+    mediaUrl = getFileFromSources(item);
+  }
```
Comment (dgirardi) [inline modules/jwplayerRtdProvider.js:229]:
```suggestion
const mediaUrl = item.file ?? getFileFromSources(item);
```

File: modules/jwplayerRtdProvider.js [hunk 3]
```diff
@@ -329,15 +353,30 @@ export function addOrtbSiteContent(ortb2, contentId, contentData) {
     content.id = contentId;
   }
 
-  const currentData = content.data = content.data || [];
+  if (contentUrl) {
+    content.url = contentUrl;
```
Comment (dgirardi) [inline modules/jwplayerRtdProvider.js:357]: Should this override what might be in `content` already?

Comment (karimMourra) [reply modules/jwplayerRtdProvider.js:357]: Yes! If a publisher has requested that we enrich the bid request for a given media, then they have determined that the bid request is specific to that media, therefore we should override existing data in that field. 

Unfortunately the oRTB spec only has room for 1 content description, which is unrealistic given that multiple videos could be rendering simultaneously on the page.

**Review by dgirardi — APPROVED** [review]

**Comment by karimMourra** [comment]
@patmmccann what does the "on hold" label mean ? Does it block this from getting merged; If so, how can we resolve?

**Comment by patmmccann** [comment]
Yes this blocked from merging, it is a breaking change

**Review by patmmccann — CHANGES_REQUESTED** [review]
This is a breaking change, please require new configuration to enable overwriting existing fields

File: modules/jwplayerRtdProvider.js (hunk 3 above)
Comment (karimMourra) [reply modules/jwplayerRtdProvider.js:370]: @patmmccann and @dgirardi this has been address by adding configuration params

File: modules/jwplayerRtdProvider.js [hunk 4]
```diff
@@ -71,15 +81,31 @@ export function fetchTargetingInformation(jwTargeting) {
   });
 }
 
+export function setOverrides(params) {
+  // For backwards compatibility, default to always unless overridden by Publisher.
+  overrideContentId = sanitizeOverrideParam(params.overrideContentId, ENRICH_ALWAYS);
```
Comment (dgirardi) [inline modules/jwplayerRtdProvider.js:86]: Assuming this should be updated for 9, do you have a way to track it - would a TODO comment be helpful?

**Review by dgirardi — APPROVED** [review]

**Review by patmmccann — APPROVED** [review]

File: modules/jwplayerRtdProvider.js (hunk 4 above)
Comment (karimMourra) [reply modules/jwplayerRtdProvider.js:88]: added!

**Comment by karimMourra** [comment]
@patmmccann can we remove the `on hold` and merge ?

**Pull request merged**